### DISCUSSION TITLE
PC XT 8088 with EMS, XMS memory?

### DISCUSSION THREAD

**Discussion by n1tr0ms** [86Box/86Box#4526] [Q&A] [answered]
Hello.

Some time ago I was interested in the idea to make a maximum PC XT build with 8088.

In the end I got this configuration:
1) 8088, 16 Mhz, Xi8088 board, 1024KB memory.
2) Original DOS 6.22 and Windows 3.0 with PVGA1A drivers, 640x480, 256 colors.
3) Cute Mouse 2.0 for DOS and Microsoft Mouse 7 mouse.drv for Windows 3.0 for correct mouse work.
4) Working Sound Blaster 1.0 and MT-32 sound cards. {Didn't find a way to check Adlib, CMS and Tandy PSSJ}
5) Working SCSI CD-ROM.
6) Working UMB via USE!UMBS.SYS.
7) 503MB HDD IDE drive.

The last aspect that I could not figure out was the EMS and XMS memory.
As far as I understood via information on the net, EMS is supported by XT, but without specifying the processor. Among emulated expansion boards the most suitable is BOCARAM and Lo-Tech EMS Board, available physically as clones or replicas, for the latter there is even a modification to add 4MB of memory. Despite this, all memory types except XMS are shown as available, but no device or program which can read and/or use them correctly (DOSMAX, EMS2UMB.EXE, UMBEMS.SYS, EMM.SYS (Windows 2.11 286), HRAM, FDXMS286.SYS, EMM286.EXE), while the main 1MB one works quite well.
I would like to know if it is possible to utilize the available memory over 1MB or is it currently impossible? Also want to know about possibilities to convert or adapt all or part of EMS into XMS or UMB.

I provide the build itself, screenshots of tests in some benchmarks and video demonstration of OS booting with information about allegedly available memory:
https://cloud.mail.ru/public/nM4F/3F7DY6ESw
https://cloud.mail.ru/public/1oNJ/MrAV554rs
https://github.com/86Box/86Box/assets/103135348/7179b186-2479-480c-acc9-1debb5ab8255

**Comment by ppgrainbow** [comment]
Can I have a screenshot of your configuration for proof?

As far as I know, 808x and 8018x processors can only address up to 1 MB of RAM leaving 640 KB of RAM usable to the end user and 384 KB for upper memory blocks.. It is possible to use EMS memory on these computers that have such processors as well as a 80286.

**Reply by n1tr0ms** [reply]
Yes.
![86Box_XT_8088](https://github.com/86Box/86Box/assets/103135348/902bf410-dc09-48dc-b28a-cb2383ec1a67)
![Monitor_1_20240611-080626-926](https://github.com/86Box/86Box/assets/103135348/9e1dab79-b114-4c15-9960-9a9df766d889)
![Monitor_1_20240611-080643-953](https://github.com/86Box/86Box/assets/103135348/3c52ae3f-df2b-4c7f-8450-eb429e0f18e3)

**Reply by jriwanek** [reply]
Yeah, you've assigned an XMS+EMS board designed for a 286 to an XT, this is an unfortunate side effect of how our config handlers currently work, but you can ONLY use that EV159 for XMS on an AT, NOT an XT, so you've picked an invalid configuration which will NEVER work, in emulation, or on real hardware.

**Reply by n1tr0ms** [reply]
What about BOCARAM and Lo-tech EMS? Could they be used in 8088 XT? I do use a modified ltemm(lthck) for the latter, but probably need to specify more parameters for identification along with /p and /i.

**Reply by n1tr0ms** [reply]
Nevermind now, I turned out what I had a wrong addressing for some devices and after re-setting LTHCK discovers the board and enabled EMS memory but CD-ROM and UMB stopped working. USE!UMBS.SYS crashes the system (hangs or screen garbage) when selecting other seemingly free addresses. And the funny thing is that lags has appear, and free memory in this case is not so much more than without EMS - 560-563kb instead of 554kb. Not a very equal exchange for possibility to play Dune II, easy to make menu in config.sys and make two options at least to run with or without CD drivers. But thanks for the response anyway.

**Reply by jriwanek** [reply]
They can, but they're still not gonna do the impossible and give you XMS on an XT class system. On an 8088, your choices are conventional, and EMS, with some, UMB, but you are *NEVER* going to have support for XMS on that class of system.

**Comment by jriwanek** [comment, marked as answer]
There were emulations which provided XMS on 286 class systems, possibly even XT class systems, but *NATIVE* XMS on an XT is quite impossible due to how the 8088 works.
